src/contracts/middleware/IndexRegistry.sol

```diff
@@ -130,25 +123,33 @@ contract IndexRegistry is IIndexRegistry {
     function getTotalOperatorsForQuorumAtBlockNumberByIndex(uint8 quorumNumber, uint32 blockNumber, uint32 index) external view returns (uint32){
         OperatorIndexUpdate memory operatorIndexToCheck = _totalOperatorsHistory[quorumNumber][index];
 
-        // blocknumber must be before the "index'th" entry's toBlockNumber
-        require(blockNumber <= operatorIndexToCheck.toBlockNumber, "IndexRegistry.getTotalOperatorsForQuorumAtBlockNumberByIndex: provided index is too far in the future for provided block number");
+        // blocknumber must be before the "index'th" entry's fromBlockNumber
+        require(blockNumber >= operatorIndexToCheck.fromBlockNumber, "IndexRegistry.getTotalOperatorsForQuorumAtBlockNumberByIndex: provided index is too far in the past for provided block number");
         
-        // if there is an index update before the "index'th" update, the blocknumber must be after the previous entry's toBlockNumber
-        if (index != 0){
-            OperatorIndexUpdate memory previousOperatorIndex = _totalOperatorsHistory[quorumNumber][index - 1];
-            require(blockNumber > previousOperatorIndex.toBlockNumber, "IndexRegistry.getTotalOperatorsForQuorumAtBlockNumberByIndex: provided index is too far in the past for provided block number");
+        // if there is an index update after the "index'th" update, the blocknumber must be before the previous entry's fromBlockNumber
+        if (index != _totalOperatorsHistory[quorumNumber].length - 1){
+            OperatorIndexUpdate memory previousOperatorIndex = _totalOperatorsHistory[quorumNumber][index + 1];
+            require(blockNumber < previousOperatorIndex.fromBlockNumber, "IndexRegistry.getTotalOperatorsForQuorumAtBlockNumberByIndex: provided index is too far in the future for provided block number");
```

Comment:
OK, just so we're on the same page -- the canonical number of operators "at a block" is the number of operators at the last update for the block.
Makes sense to me; we should do what we can to make this crystal clear

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/81/commits/01f331ade2b5ca39933d03828ab4ba0cab624766